File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int max = 0;
        for(int p : piles){
            max = Math.max(max, p);
        }

        int left = 1;
        int right = max;
        // int min = max;

        while (left <= right){
            int mid = left + (right - left) / 2;
            long k = 0;
            for(int p : piles){
                k += (p + mid -1) / mid;
            }
            if(k <= h){
                right = mid - 1;
            }
            else left = mid + 1;
        }
        return left;
    }
}

File: Data Structures & Algorithms/jump-game-vii/submission-0.py
class Solution:
    def canReach(self, s: str, minJump: int, maxJump: int) -> bool:
        if s[-1] != '0': return False
        queue = deque([0])
        farthest = 0
        n = len(s)
        
        while queue:
            i = queue.popleft()
            left = max(farthest + 1, i + minJump)
            right = min(n-1, i + maxJump)

            for j in range(left, right+1):
                if s[j] == '0':
                    if j == n-1:
                        return True
                    queue.append(j)
                farthest = right
        return False